kopykat-annotations/src/main/kotlin/at/kopyk/Copy.kt

```diff
@@ -0,0 +1,14 @@
+package at.kopyk
+
+import kotlin.reflect.KClass
+
+/**
+ * Marks type or typealias to generate copy constructors.
+ * This is the equivalent of using [CopyFrom] and [CopyTo] together.
+ *
+ * Both the subject of this annotation and the provided type
+ * have to have the same constructor properties.
+ */
+@Target(AnnotationTarget.CLASS, AnnotationTarget.TYPEALIAS)
+@Retention(AnnotationRetention.SOURCE)
+public annotation class Copy(val type: KClass<*>)
```

Comment:
Could we have these three classes in a single file? If you prefer to keep it this way it's also fine, of course.